src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,42 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getTodo, getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  todoId: number;
+  setShownTodoId: (value: null | number) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todoId, setShownTodoId }) => {
+  const [loading, setLoading] = useState(true);
+  const [todo, setTodo] = useState<null | Todo>(null);
+  const [user, setUser] = useState<null | User>(null);
```

Comment:
you should indicate primary type as the first one (as far as you expect it, not `null`)  
```suggestion
  const [todo, setTodo] = useState<Todo | null>(null);
  const [user, setUser] = useState<User | null>(null);
```